data/src/main/java/com/microsoft/azure/kusto/data/UriUtils.java

```diff
@@ -67,4 +67,22 @@ public static String[] getSasAndEndpointFromResourceURL(String url) throws URISy
         }
         return parts;
     }
+
+    public static String stripFileNameFromCommandLine(String cmdLine) {
+        String processNameForTracing = cmdLine;
+
+        if (processNameForTracing != null) {
+            String[] splitCommand = processNameForTracing.split(" ")[0].split("\\.");
+            processNameForTracing = splitCommand[splitCommand.length - 1];
+            if (processNameForTracing.equalsIgnoreCase("jar") && splitCommand.length > 1) {
+                processNameForTracing = splitCommand[splitCommand.length - 2];
+            }
+            int lastDirSeparator = processNameForTracing.lastIndexOf(File.separatorChar);
+            if (lastDirSeparator > 0) {
+                processNameForTracing = processNameForTracing.substring(lastDirSeparator + 1);
+            }
+        }
+
+        return processNameForTracing;
```

Comment:
```suggestion
        if (cmdLine == null) {
            return null;
        }

        String[] splitCommand = cmdLine.split(" ")[0].split("\\.");
        String processNameForTracing = splitCommand[splitCommand.length - 1];
        if (processNameForTracing.equalsIgnoreCase("jar") && splitCommand.length > 1) {
            processNameForTracing = splitCommand[splitCommand.length - 2];
        }
        int lastDirSeparator = processNameForTracing.lastIndexOf(File.separatorChar);
        if (lastDirSeparator > 0) {
            processNameForTracing = processNameForTracing.substring(lastDirSeparator + 1);
        }

        return processNameForTracing;
```

I think you can reduce the nesting here and make it more readable